docs/two-phase-commit-transactions.md

```diff
@@ -263,113 +347,409 @@ interface ServiceB {
+{% endcapture %}
+
+<div class="notice--warning">{{ notice--warning | markdownify }}</div>
+
+For instance, in the example code in [Execute a transaction by using multiple transaction manager instances](#execute-a-transaction-by-using-multiple-transaction-manager-instances), multiple transaction managers (`transactionManager1` and `transactionManager2`) are used in a single process for ease of explanation. However, that example code doesn't include a way to handle exceptions.
+
+The following example code shows how to handle exceptions in transactions with a two-phase commit interface:
+
+```java
+public class Sample {
+  public static void main(String[] args) throws Exception {
+    TransactionFactory factory1 =
+        TransactionFactory.create("<PATH_TO_CONFIGURATION_FILE_FOR_TRANSACTION_MANAGER_1>");
+    TwoPhaseCommitTransactionManager transactionManager1 =
+        factory1.getTwoPhaseCommitTransactionManager();
+
+    TransactionFactory factory2 =
+        TransactionFactory.create("<PATH_TO_CONFIGURATION_FILE_FOR_TRANSACTION_MANAGER_2>");
+    TwoPhaseCommitTransactionManager transactionManager2 =
+        factory2.getTwoPhaseCommitTransactionManager();
+
+    int retryCount = 0;
+    TransactionException lastException = null;
+
+    while (true) {
+      if (retryCount++ > 0) {
+        // Retry the transaction three times maximum in this sample code.
+        if (retryCount >= 3) {
+          // Throw the last exception if the number of retries exceeds the maximum.
+          throw lastException;
+        }
+
+        // Sleep 100 milliseconds before retrying the transaction in this sample code.
+        TimeUnit.MILLISECONDS.sleep(100);
+      }
+
+      TwoPhaseCommitTransaction transaction1 = null;
+      TwoPhaseCommitTransaction transaction2 = null;
+      try {
+        // Begin a transaction.
+        transaction1 = transactionManager1.begin();
+
+        // Join the transaction that `transactionManager1` begun by using the transaction ID.
+        transaction2 = transactionManager2.join(transaction1.getId());
+
+        // Execute CRUD operations in the transaction.
+        Optional<Result> result = transaction1.get(...);
+        List<Result> results = transaction2.scan(...);
+        transaction1.put(...);
+        transaction2.delete(...);
+
+        // Prepare the transaction.
+        prepare(transaction1, transaction2);
+
+        // Validate the transaction.
+        validate(transaction1, transaction2);
+
+        // Commit the transaction.
+        commit(transaction1, transaction2);
+      } catch (UnsatisfiedConditionException e) {
+        // You need to handle `UnsatisfiedConditionException` only if a mutation operation specifies 
+        // a condition. This exception indicates the condition for the mutation operation is not met.
+
+        rollback(transaction1, transaction2);
+
+        // You can handle the exception here, according to your application requirements.
+
+        return;
```

Comment:
This exception is part of a [feature](https://github.com/scalar-labs/scalardb/pull/899) for the future ScalarDB 4.0 so we can remove it.
```suggestion

```